src/ui/MainWindow.cpp

```diff
@@ -97,6 +98,7 @@ bool MainWindow::showUI()
     qml->rootContext()->setContextProperty(QLatin1String("torControl"), torControl);
     qml->rootContext()->setContextProperty(QLatin1String("torInstance"), Tor::TorManager::instance());
     qml->rootContext()->setContextProperty(QLatin1String("uiMain"), this);
+    qml->rootContext()->setContextProperty(QLatin1String("languagesModel"), new LanguagesModel(this));
```

Comment:
You should use `qmlRegisterType` instead of context properties, like ContactsModel does. Then, in LanguagePreferences.qml, you can add `import im.ricochet 1.0` and create a `LanguagesModel` as-needed.